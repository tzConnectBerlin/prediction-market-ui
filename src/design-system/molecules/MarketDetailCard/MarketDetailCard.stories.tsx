import * as React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';
import { MarketDetailCard, MarketDetailCardProps } from './MarketDetailCard';

export default {
  title: 'Molecule/MarketDetailCard',
  component: MarketDetailCard,
  argTypes: {},
} as Meta;

const descriptionArgs = {
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.'.repeat(20),
  expandActionText: 'Read more',
  shrinkActionText: 'Read less',
};

const Template: Story<MarketDetailCardProps> = (args) => <MarketDetailCard {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: 'About Market',
  items: [
    {
      title: 'Description',
      item: descriptionArgs,
    },
    {
      title: 'Ticker',
      item: '$TZBDN',
    },
  ],
};
