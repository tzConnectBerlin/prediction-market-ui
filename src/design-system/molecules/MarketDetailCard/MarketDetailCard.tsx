import * as React from 'react';
import styled from '@emotion/styled';
import { Paper, Grid, useTheme, useMediaQuery } from '@material-ui/core';
import { Typography } from '../../atoms/Typography';
import { ExpandText, ExpandTextProps } from '../ExpandText/ExpandText';

interface MarketCardItem {
  title: string;
  item: string | ExpandTextProps;
}

export interface MarketDetailCardProps {
  title: string;
  items: MarketCardItem[];
}

const PaperWrapperStyled = styled(Paper)`
  padding: 2rem;
`;

export const MarketDetailCard: React.FC<MarketDetailCardProps> = ({ title, items }) => {
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
  return (
    <PaperWrapperStyled>
      <Typography size="h2" fontWeight="bold">
        {title}
      </Typography>
      <Grid container direction="column" alignContent="flex-start" spacing="2" pb="1rem">
        {items.map((data, index) => {
          const key = typeof data.item === 'string' ? data.item : data.item.text;
          return (
            <Grid item mt="2rem" key={`${key}-${index}`} width="100%">
              <Typography size="body1" fontWeight="bold" mb="0.75rem">
                {data.title}
              </Typography>
              {typeof data.item === 'string' && data.title === 'Adjudicator' && isMobile ? (
                <Typography color="primary">{`${data.item?.substring(
                  0,
                  10,
                )}...${data.item?.substring(data.item?.length - 10)}`}</Typography>
              ) : typeof data.item === 'string' ? (
                data.title === 'Adjudicator' ? (
                  <Typography color="primary">{data.item}</Typography>
                ) : (
                  <Typography>{data.item}</Typography>
                )
              ) : (
                <ExpandText {...data.item} />
              )}
            </Grid>
          );
        })}
      </Grid>
    </PaperWrapperStyled>
  );
};
