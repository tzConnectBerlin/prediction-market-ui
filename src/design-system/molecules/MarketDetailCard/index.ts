export { MarketDetailCard, MarketDetailCard as default } from './MarketDetailCard';
export type { MarketDetailCardProps } from './MarketDetailCard';
