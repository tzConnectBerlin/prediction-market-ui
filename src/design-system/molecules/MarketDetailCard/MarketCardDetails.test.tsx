import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { MarketDetailCard } from './MarketDetailCard';

const descriptionArgs = {
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.'.repeat(20),
  expandActionText: 'Read more',
  shrinkActionText: 'Read less',
};

const defaultProps = {
  title: 'About Market',
  items: [
    {
      title: 'Description',
      item: descriptionArgs,
    },
    {
      title: 'Ticker',
      item: '$TZBDN',
    },
  ],
};

describe('Snapshot testing MarketDetailCard Component', () => {
  it('renders correctly default data', () => {
    const marketDetailCard = renderer.create(<MarketDetailCard {...defaultProps} />).toJSON();
    expect(marketDetailCard).toMatchSnapshot();
  });
});

describe('Element testing MarketDetailCard Component', () => {
  it('render correctly with given data', async () => {
    const { getByText } = render(<MarketDetailCard {...defaultProps} />);
    expect(getByText(/Description/i)).toBeInTheDocument();
    expect(getByText(/Ticker/i)).toBeInTheDocument();
  });
});
