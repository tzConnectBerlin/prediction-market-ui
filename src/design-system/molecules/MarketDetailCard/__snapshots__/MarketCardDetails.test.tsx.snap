// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing MarketDetailCard Component renders correctly default data 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  padding: 2rem;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
  font-weight: 700;
}

.emotion-1.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: calc(100% + 2px);
  margin-top: -2px;
  margin-left: -2px;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  padding: 1rem;
}

.emotion-2>.MuiGrid-item {
  max-width: none;
}

.emotion-2>.MuiGrid-item {
  padding-top: 2px;
  padding-left: 2px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  margin-top: 1rem;
  width: 100%;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
  color: #3f51b5;
}

.emotion-4.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-5 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-5.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-6 {
  margin: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.12);
  border-bottom-width: thin;
  margin-top: 1rem;
}

<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-0"
>
  <h5
    className="MuiTypography-root MuiTypography-h5  emotion-1"
    lines={3}
  >
    About Market
  </h5>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column emotion-2"
  >
    <div
      className="MuiGrid-root MuiGrid-item emotion-3"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6  emotion-4"
        lines={3}
      >
        Description
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body1  truncate emotion-5"
        lines={3}
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
      <hr
        className="MuiDivider-root MuiDivider-fullWidth emotion-6"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item emotion-3"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6  emotion-4"
        lines={3}
      >
        Ticker
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body1  emotion-5"
        lines={3}
      >
        $TZBDN
      </p>
    </div>
  </div>
</div>
`;
